plugins/scaffolder-backend-module-github/src/actions/githubDeployKey.ts

```diff
@@ -126,20 +126,35 @@ export function createGithubDeployKeyAction(options: {
 
       const client = new Octokit(octokitOptions);
 
-      await client.rest.repos.createDeployKey({
-        owner: owner,
-        repo: repo,
-        title: deployKeyName,
-        key: publicKey,
+      await ctx.checkpoint({
+        key: `create.deploy.key.${owner}.${repo}.${publicKey}`,
+        fn: async () => {
+          await client.rest.repos.createDeployKey({
+            owner: owner,
+            repo: repo,
+            title: deployKeyName,
+            key: publicKey,
+          });
+        },
       });
-      const publicKeyResponse = await client.rest.actions.getRepoPublicKey({
-        owner: owner,
-        repo: repo,
+
+      const { key, keyId } = await ctx.checkpoint({
```

Comment:
Does this even need to be a checkpoint? 

This is just getting the key right? No harm in re-running this?

Reply:
True, no harm, I was thinking also to reduce unnecessary calls to a provider, if we can cache them.
Then re-run will be much faster